csrc/device_lower/validation.cpp

```diff
@@ -876,10 +876,14 @@ namespace {
 //!  specialization of tidx as lane id.
 void validateMmaTensors(MmaOp* mma) {
   bool tidx_validated = false;
-  std::vector<TensorView*> to_validate = {
-      mma->inA()->as<TensorView>(),
-      mma->inB()->as<TensorView>(),
-      mma->out()->as<TensorView>()};
+  std::vector<TensorView*> to_validate = {mma->out()->as<TensorView>()};
```

Comment:
It appears to me that `MmaOp::out`, `MmaOp::inA` and `MmaOp::inB` always return a `TensorView`. I suggest (in a separate PR) we make the return type `TensorView*` instead of `Val*`.

Reply:
They can actually be `kir::TensorIndex` as well, for example, in https://github.com/NVIDIA/Fuser/blob/d243121980debe620caea898801336928ea4a121/csrc/codegen.cpp#L1151, it should return a `kir::TensorIndex`. Generally, for tensor op, there in order to know if its input or output is always a `TensorView`, you can look at https://github.com/NVIDIA/Fuser/blob/d243121980debe620caea898801336928ea4a121/csrc/device_lower/pass/fusion_simplifier.cpp#L20 and https://github.com/NVIDIA/Fuser/blob/main/csrc/device_lower/pass/index.cpp, if it is replaced with a `LoadStoreOp` in `UnaryOpInserter` or if it is lowered as a `LoadStoreOp` in indexing lowering, then the operands and outputs of such an expr must be `TensorView`. Otherwise, it can also be `kir::TensorIndex`.